content/en/blog/_posts/2019-06-05-what-and-the-why-of-cluster-api.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: blog
+title: 'The What and the Why of the Cluster API'
+date: 2019-06-05
+---
+
+**Author**: Tim St. Clair, Senior Staff Engineer at VMware
+
+_Note: This post was originally published on the [VMware Cloud Native Apps blog](https://blogs.vmware.com/cloudnative/2019/03/14/what-and-why-of-cluster-api/) and republished here with author's permission_
+
+Throughout the evolution of software tools there exists a tension between generalization and partial specialization. A tool’s broader adoption is a form of natural selection, where its evolution is predicated on filling a given need, or role, better than its competition. This premise is imbued in the central tenets of Unix philosophy:
+
+* **Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new features.**
+* **Expect the output of every program to become the input to another, as yet unknown, program.**
+
+
+The domain of configuration management tooling is rife with examples of not heeding this lesson (i.e. Terraform, Puppet, Chef, Ansible, Juju, Saltstack, etc), where expanse in generality has given way to partial specialization of different tools, causing fragmentation of an ecosystem. This pattern has not gone unnoticed by those in the Kubernetes cluster lifecycle special interest group, or SIG, whose objective is to simplify the creation, configuration, upgrade, downgrade, and teardown of Kubernetes clusters and their components. Therefore, one of the primary design principles for any subproject that the SIG endorses is: **Where possible, tools should be composable to solve a higher order set of problems.**
+
+## Background
+
+Early in the evolution of the Kubernetes project, there was a desire for configurability, as different environments had varying constraints. This flexibility gave way to a myriad of assertions, and opinions, that initially fragmented the community around installation paths. However, it was clear to the community that there were a set of common overlapping concerns, or a lowest common denominator. In order to address these specific concerns, [kubeadm](https://github.com/kubernetes/kubeadm) was born as a composable bootstrapping tool for Kubernetes clusters.
```

Comment:
How about hyperlinking “kubeadm” to the documentation page for the tool? Would that tweak be OK?

Reply:
sure, that works too. will change it, hopefully the author will be okay with it ;)

Reply:
@sftim how does it look ? 

I was thinking to chose https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm/ but then i ended up with the setup one - some reorg of the content will be welcome to bring some clarity ...things are spread everywhere